testing/integration/upgrade_test.go

```diff
@@ -273,7 +274,165 @@ func TestStandaloneUpgrade(t *testing.T) {
 	require.NoError(t, err, "error connecting client to agent")
 	defer c.Disconnect()
 
-	_, err = c.Upgrade(ctx, upgradeInputVersion.String(), "", false)
+	_, err = c.Upgrade(ctx, upgradeInputVersion.String(), "", false, false)
+	require.NoErrorf(t, err, "error triggering agent upgrade to version %q", upgradeInputVersion.String())
+
+	require.Eventuallyf(t, func() bool {
+		state, err := c.State(ctx)
+		if err != nil {
+			t.Logf("error getting the agent state: %v", err)
+			return false
+		}
+		t.Logf("current agent state: %+v", state)
+		return state.Info.Commit == expectedAgentHashAfterUpgrade && state.State == cproto.State_HEALTHY
+	}, 5*time.Minute, 1*time.Second, "agent never upgraded to expected version")
+
+	checkUpgradeWatcherRan(t, agentFixture)
+
+	version, err := c.Version(ctx)
+	require.NoError(t, err, "error checking version after upgrade")
+	require.Equal(t, expectedAgentHashAfterUpgrade, version.Commit, "agent commit hash changed after upgrade")
+}
+
+func TestStandaloneUpgrade_FailInitialCheck(t *testing.T) {
+	define.Require(t, define.Requirements{
+		// Stack:   &define.Stack{},
+		Local:   false, // requires Agent installation
+		Isolate: true,
+		Sudo:    true, // requires Agent installation
+	})
+
+	agentFixture, err := define.NewFixture(t, define.Version())
+
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	err = agentFixture.Prepare(ctx)
+	require.NoError(t, err, "error preparing agent fixture")
+
+	err = agentFixture.Configure(ctx, []byte(fastWatcherCfg))
+	require.NoError(t, err, "error configuring agent fixture")
+
+	const minVersionString = "8.10.0-SNAPSHOT"
+	minVersion, _ := version.ParseVersion(minVersionString)
```

Comment:
I know its probably not going to happen, but just to be sure that in the case that `version.ParseVersion` throws an error it is checked. Online an extra line.

`require.NoError(t, err)`